### PR TITLE
CQ-4345761 Implementation of clientLibRef

### PR DESCRIPTION
**Pull request by rismehta**
- Also added schemaRef and schemaType which is required for editor @review @vdua
DOD(Yes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java [hunk 1]
```diff
@@ -64,6 +70,24 @@ public String getThankYouMessage() {
...
     }
 
+    @Override
+    @Nullable
+    public String getClientLibRef() {
+        return clientLibRef;
+    }
+
+    @Override
+    @Nullable
```
Comment (nit23uec) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:80]: Author view?

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java [hunk 2]
```diff
@@ -64,6 +70,24 @@ public String getThankYouMessage() {
...
+    public String getSchemaRef() {
+        return GuideContainer.from(resource).getSchemaRef();
+    }
+
+    @Override
+    @Nullable
+    public GuideSchemaType getSchemaType() {
```
Comment (nit23uec) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:87]: Author view?

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java [hunk 3]
```diff
@@ -64,6 +70,24 @@ public String getThankYouMessage() {
         return thankyouMessage;
     }
 
+    @Override
+    @Nullable
+    public String getClientLibRef() {
```
Comment (nit23uec) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:75]: `@JsonIgnore`?

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/util/ComponentUtils.java [hunk 4]
```diff
@@ -50,6 +52,17 @@ public static String getEncodedPath(@NotNull String path) {
...
 
+    /**
+     * Checks if the given resource if a core adaptive form container
+     * 
+     * @param resource reference to the {@link Resource}
+     * @return true, if it is an adaptive form container, false otherwise
+     */
+    @NotNull
+    public static boolean isAFContainer(@NotNull Resource resource) {
+        return resource.isResourceType(RT_FD_FORM_CONTAINER_V2);
```
Comment (nit23uec) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/util/ComponentUtils.java:63]: pls use `fieldType` check.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/page/v1/page/.content.xml [hunk 5]
```diff
@@ -19,5 +19,5 @@
           jcr:primaryType="cq:Component"
           cq:icon="page"
           jcr:title="Adaptive Form Core Page"
-          sling:resourceSuperType="core/wcm/extensions/amp/components/page/v1/page"
+          sling:resourceSuperType="core/wcm/components/page/v3/page"
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/page/v1/page/.content.xml:22]: is this desired?

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java (hunk 1 above)
Comment (rismehta) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:80]: We would need this in author view for content tree use-case.

**Pull request closed without merging**